src/installer/tests/Assets/TestProjects/HostApiInvokerApp/HostFXR.cs

```diff
@@ -142,6 +194,57 @@ static void Test_hostfxr_set_error_writer(string[] args)
             }
         }
 
+        /// <summary>
+        /// Test that invokes native api hostfxr_get_dotnet_environment_info.
+        /// </summary>
+        /// <param name="args[0]">hostfxr_get_dotnet_environment_info</param>
+        /// <param name="args[1]">(Optional) Path to the directory with dotnet.exe</param>
+        static void Test_hostfxr_get_dotnet_environment_info(string[] args)
+        {
+            string? dotnetExeDir = null;
+            if (args.Length >= 2)
+                dotnetExeDir = args[1];
+
+            string hostfxr_version;
+            string hostfxr_commit_hash;
+            List<hostfxr.hostfxr_dotnet_environment_sdk_info> sdks = new List<hostfxr.hostfxr_dotnet_environment_sdk_info>();
+            List<hostfxr.hostfxr_dotnet_environment_framework_info> frameworks = new List<hostfxr.hostfxr_dotnet_environment_framework_info>();
+
+            int rc = hostfxr.hostfxr_get_dotnet_environment_info(
+                dotnet_root: dotnetExeDir,
+                reserved: IntPtr.Zero,
+                result: (info, result_context) => {
+                    hostfxr_version = info.hostfxr_version;
+                    hostfxr_commit_hash = info.hostfxr_commit_hash;
+                    for (int i = 0; i < info.sdk_count; i++)
+                    {
+                        IntPtr pSdkInfo = new IntPtr(info.sdks.ToInt64() + (i * Marshal.SizeOf<hostfxr.hostfxr_dotnet_environment_sdk_info>()));
```

Comment:
This should validate the `size` fields - it's a test, so it needs to make sure that all the fields are set propertly.